### DISCUSSION TITLE
Welcome to bandersnatch Discussions!

### DISCUSSION THREAD

**Discussion by hongaar** [hongaar/bandersnatch#170] [Show and tell]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by khill-fbmc** [comment]
Idea... Is it possible to modify the terminal prompt when running in REPL mode?

My idea is to be able to persist state somehow and use commands to set some sort of static value...

If I was making a network tool...

```shell
Input command (explicit)

> ping tacobell.com

... output ...

> set-host tacobell.com

 (tacobell.com) > ping

... same output, no prompt for host (implicit) ...

 (tacobell.com) > dig

... output of the `dig tacobell.com` command (implicit) ...
```

**Reply by khill-fbmc** [reply]
Or more simply, a different character? hehe

**Reply by hongaar** [reply]
This should be possible with a call to `program.options.prompt = 'some new prompt >'`

See https://github.com/hongaar/bandersnatch/blob/main/examples/repl.ts#L33

However, there seems to be some kind of bug with the REPL server. You need to type `Space + Backspace` in the new prompt to see this behaviour. Not sure what is causing this. :thinking:
